### ISSUE TITLE
Since fastify 4, fastify no longer waits for existing requests to finish when calling fastify.close()

### ISSUE THREAD

**Issue by corevo** [fastify/fastify#4098]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the regression has not already been reported

### Last working version

3.29.0

### Stopped working in version

4.0.0

### Node.js version

18.4.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.4

### 💥 Regression Report

As of fastify 4, fastify does not wait for requests made before `instance.close` was called to finish, rather it shuts down immediately and hangs up the connection.

### Steps to Reproduce

A repo that repros the issue is available [here](https://github.com/corevo/fastify-shutdown-regression)

Steps to reproduce the issue:

```sh
$ git clone git@github.com:corevo/fastify-shutdown-regression.git
$ cd fastify-3
$ npm i
$ npm t
# should exit with 'OK' and exit code 0
$ cd fastify-4
$ npm i
$ npm t
# will instead fail on socket hang up
```

### Expected Behavior

When calling `instance.close` fastify should wait up to a certain (ideally configurable) grace period, for any request that has already been made prior to closing to finish.

**Comment by corevo** [comment]
I've realized what the issue was, I needed to set `forceCloseConnections: false`

**Comment by mcollina** [comment]
@ShogunPanda could you take a look? There should not be a visible change of behavior.

**Comment by xtx1130** [comment]
I think this question can be closed, I can't reproduce it in fastify 4.2.0

**Comment by climba03003** [comment]
> I think this question can be closed, I can't reproduce it in fastify 4.2.0

I didn't see any commit that would potentially fix the issue. Will it be related to `node` itself?

https://github.com/fastify/fastify/compare/v4.0.0...v4.2.0

**Comment by ShogunPanda** [comment]
Confirmed, the bug is in Node.js and the PR above fixed it. The fix should come out in 18.7.0.

@mcollina Shall we close this?
